### PR TITLE
feat: Update cluster defaults package with network policies

### PR DESCRIPTION
**Pull request by borkodjurkovic-ssc**
# Summary

In order to comply with the `nist-sp-800-53-r5-require-namespace-network-policies` constraint (of [NIST SP 800-53 Rev. 5 Policy Controller bundle](https://github.com/GoogleCloudPlatform/gke-policy-library/tree/main/anthos-bundles/nist-sp-800-53-r5)), `cluster-defaults` package required updates to add network policies to the `gateway-infra` and `default` namespaces.

## `gateway-infra` namespace Network Policy
Added `cluster-defaults/admin-namespaces/networkpolicy.yaml` file to implement network policy in the `gateway-infra` namespace.

Network policies implement following rules:
- Allow ingress within namespace
- Allow ingress from lb health check
- Allow egress within namespace
- Allow egress to metadata server
- Allow egress for GCP API
- Allow egress to private IP ranges (includes K8S cluster)

## `default` namespace Network Policy
Added `cluster-defaults/default-namespace/networkpolicy.yaml` file to implement network policy in the `default` namespace

Network policies implement following rules:
- Allow ingress within namespace
- Allow egress within namespace

### CODE REVIEW

**Review by fmichaelobrien — DISMISSED** [review]
Lgtm

**Comment by fmichaelobrien** [comment]
Doing a deeper review around general 800-53 later tonight

File: solutions/gke/kubernetes/cluster-defaults/admin-namespaces/networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,149 @@
+# Copyright 2021 Google LLC
```
Comment (alaincormier-ssc) [inline solutions/gke/kubernetes/cluster-defaults/admin-namespaces/networkpolicy.yaml:1]: the documentation (README.md and securitycontrols.md) should be re-generated

**Review by alaincormier-ssc — APPROVED** [review]

**Review by davelanglois-ssc — APPROVED** [review]

**Review by fmichaelobrien — APPROVED** [review]
/lgtm

**Pull request merged**